### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by i-stam**

### CODE REVIEW

File: contracts/ReversibleICO.sol [hunk 1]
```diff
@@ -782,23 +863,26 @@ contract ReversibleICO is IERC777Recipient {
                     }
                 }
 
+                // return overflow tokens received
                 if(returnTokenAmount > 0) {
-                    // return overflow tokens received
-
-                    // allocate tokens to participant
+                    // send tokens back participant
```
Comment (mickys) [inline contracts/ReversibleICO.sol]: // send tokens back participant
>>
// send tokens back to participant

File: contracts/ReversibleICO.sol [hunk 2]
```diff
@@ -322,13 +357,19 @@ contract ReversibleICO is IERC777Recipient {
...
-        require(_ethAmount <= unlocked, "Requested amount to large, not enough unlocked ETH available.");
+        require(_ethAmount <= unlocked, "Requested amount too big, not enough unlocked ETH available.");
 
+        // update stats:  #project withdrawals, total amount withdrawn by the project
         projectWithdrawCount++;
         projectWithdrawnETH += _ethAmount;
 
 
+        // transfer ETH to project wallet
+        address(uint160(projectWalletAddress)).transfer(_ethAmount);
```
Comment (mickys) [inline contracts/ReversibleICO.sol]: Strange why you moved the transfer before the events,
Is there some kind of standard that says "events should be emitted" at the end of an operation ?

The only thing that comes to mind that might be affected by emitting and transferring later is resulting event order in like an app ( web3 filter ) on the resulting block.

File: contracts/ReversibleICO.sol [hunk 3]
```diff
@@ -520,13 +577,23 @@ contract ReversibleICO is IERC777Recipient {
...
+    /// @dev Returns true if the participant has locked tokens in current stage.
+    /// @param _participantAddress the address of the relevant participant.
     function canCancelByTokens(address _participantAddress) public  view  returns (bool) {
         if(getLockedTokenAmount(_participantAddress) > 0) {
             return true;
         }
         return false;
     }
 
+<<<<<<< Updated upstream
```
Comment (mickys) [inline contracts/ReversibleICO.sol]: Probably want to fix this.

File: contracts/ReversibleICO.sol [hunk 4]
```diff
@@ -782,23 +863,26 @@ contract ReversibleICO is IERC777Recipient {
...
+                // Update total ETH withdrawn
                 withdrawnETH += returnETHAmount;
 
-                // allocate remaining eth to project directly
+                // allocate remaining ETH to project directly
                 participantRecord.allocatedETH = allocatedEthAmount;
                 projectAllocatedETH = projectAllocatedETH.add(participantRecord.allocatedETH);
 
-                participantRecord.withdrawnETH += returnETHAmount;
+                participantRecord.withdrawnETH += returnETHAmount; //DUPLICATE??
```
Comment (mickys) [inline contracts/ReversibleICO.sol]: yes, duplicate removed in develop

File: contracts/ReversibleICO.sol [hunk 5]
```diff
@@ -913,50 +1000,61 @@ contract ReversibleICO is IERC777Recipient {
...
+            // update participant's audit values
             participantRecord.reservedTokens = 0;
             participantRecord.withdrawnETH += participantAvailableETH;
 
-            // globals
+
             // since this balance was never actually "accepted" it counts as returned
             // otherwise it interferes with project withdraw calculations
-            returnedETH += participantAvailableETH;
+            returnedETH += participantAvailableETH; // TO DO: REMOVE DUPLICATE!!!
```
Comment (mickys) [inline contracts/ReversibleICO.sol]: yes, already removed in develop

File: contracts/ReversibleICO.sol (hunk 2 above)
Comment (i-stam) [reply contracts/ReversibleICO.sol]: There is no standard or serious reason, it's mainly a matter of coding conventions, the only thing is that if the transfer reverts then the whole thing will revert saving a bit of gas. I think it better to execute first the statements that might revert (depending on the use case of course.

File: contracts/ReversibleICO.sol (hunk 3 above)
Comment (i-stam) [reply contracts/ReversibleICO.sol]: fixed!

File: contracts/ReversibleICO.sol (hunk 1 above)
Comment (i-stam) [reply contracts/ReversibleICO.sol]: fixed, thanks

**Pull request merged**